lua/neogit/popups/commit.lua

```diff
@@ -3,13 +3,29 @@ local status = require 'neogit.status'
 local cli = require("neogit.lib.git.cli")
 local input = require("neogit.lib.input")
 local Buffer = require("neogit.lib.buffer")
-local a = require('neogit.async')
-local uv = require('neogit.async.uv')
+local a = require 'plenary.async_lib'
+local async, await, scheduler, void, wrap = a.async, a.await, a.scheduler, a.void, a.wrap
+local uv = a.uv
 local split = require('neogit.lib.util').split
 
 local COMMIT_FILE = '.git/NEOGIT_COMMIT_EDITMSG'
 
-local get_commit_message = a.wrap(function (content, cb)
+local read_file = async(function(path)
```

Comment:
I think we should move this function into util or a separate file

Reply:
or maybe into plenary

Reply:
oh I would love that in plenary, but I think our use case is a bit too specific, since I don't want the function to error if the file doesn't exist.

Separate file does sound good though, I'll also wrap that with `git_root` while I'm at it so we can be sure our file is read from the git root when calling (which is what we want 99% of the time :sweat_smile:

Reply:
I don't think that this use case is too specific. We shouldn't be ignoring errors anyway, they could be more than the errors that we are just trying to ignore. We should be dedicating checking if the file doesn't exist to the caller. A better way to check if a file doesn't exist is to just use `uv_fs_stat` before calling the function. We could also just use the plenary path library to check but tj is refactoring it.

Reply:
> A better way to check if a file doesn't exist is to just use uv_fs_stat before calling the function.

At least in nodejs, this is considered bad practice, since the file can appear/disappear in between the calls (likely not a problem in our case, but still)

See:
https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback

> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

Reply:
That makes sense